expression/builtin_string.go

```diff
@@ -3216,18 +3225,24 @@ func (b *builtinInsertBinarySig) evalString(row chunk.Row) (string, bool, error)
 	}
 
 	if length > strLength-pos+1 || length < 0 {
+		if uint64(pos-1+int64(len(newstr)))*uint64(mysql.MaxBytesOfCharacter) > b.maxAllowedPacket {
+			b.ctx.GetSessionVars().StmtCtx.AppendWarning(errWarnAllowedPacketOverflowed.GenByArgs("insert", b.maxAllowedPacket))
+			return "", true, nil
+		}
 		return str[0:pos-1] + newstr, false, nil
 	}
 	return str[0:pos-1] + newstr + str[pos+length-1:], false, nil
```

Comment:
should we check whether `len(str[0:pos-1] + newstr + str[pos+length-1:]) > b. maxAllowedPacket ` ?